### PR TITLE
feat: add /api/status and /api/capabilities endpoints (CB-10)

### LINKED ISSUE #52: Add a status endpoint for feature flags and dependencies
## Problem
`GET /healthcheck` only returns uptime. The service now depends on several gated integrations and feature flags, so uptime alone is not enough to know whether the bot can actually process alerts.

## Proposed scope
Add a machine-readable status endpoint, for example:
- `GET /api/status`
- or `GET /api/capabilities`

Return non-sensitive metadata such as:
- service name, commit, and environment
- enabled feature flags
- enabled delivery channels
- dependency readiness / configuration status for Telegram, WhatsApp, Gemini, TradingView MCP, Firestore, and Sentry

## Why this matters
- Faster ops triage in Render preview and production environments.
- External monitors can distinguish "service is up" from "service can process alerts".
- Makes the current feature surface visible to automation and support tooling.

## Acceptance criteria
- No secrets are leaked.
- Response is cheap, stable, and JSON-only.
- The endpoint works even when some optional integrations are disabled.

### LINKED ISSUE #76: Fix provider-specific news monitor LLM readiness in status endpoint
## Finding

PR #57 adds `/api/status` / `/api/capabilities`, but the new readiness payload can report the news monitor as healthy while its primary LLM provider is not actually configured.

## Concrete evidence

- PR #57 adds `src/controllers/status.js` and computes Gemini readiness for the news monitor from `MODEL_PROVIDER === 'gemini'`, then only checks `GEMINI_API_KEY` for `dependencies.gemini.configured`.
- The same PR only reports Azure readiness under `dependencies.llmAlertEnrichment`, gated by `ENABLE_LLM_ALERT_ENRICHMENT && ENABLE_NEWS_MONITOR`; it does not report Azure/OpenRouter readiness for the primary `MODEL_PROVIDER=azure` or `MODEL_PROVIDER=openrouter` news-monitor path.
- Existing `src/controllers/webhooks/handlers/newsMonitor/analyzer.js` calls `analyzeNewsForSymbol()` for each symbol.
- Existing `src/services/grounding/gemini.js` calls `genaiClient.llmCallv2()` for the news analysis LLM call.
- Existing `src/services/grounding/genaiClient.js` dispatches `llmCallv2()` by `MODEL_PROVIDER`: Gemini is attempted only when `GEMINI_MODEL_NAME` is set, Azure only when `getAzureAIClient().validate()` passes, and OpenRouter only when `getOpenRouterClient().validate()` passes.
- Existing `src/services/inference/azureAiClient.js` requires endpoint, `AZURE_LLM_KEY`, and `AZURE_LLM_MODEL`; `src/services/inference/openRouterClient.js` requires `OPENROUTER_API_KEY` and model.

## Impact

False-green `/api/status` examples:

- `ENABLE_NEWS_MONITOR=true`, `MODEL_PROVIDER=gemini`, `GEMINI_API_KEY` set, but `GEMINI_MODEL_NAME` unset: status reports Gemini ready, while `llmCallv2()` skips Gemini and can throw `No providers configured`.
- `ENABLE_NEWS_MONITOR=true`, `MODEL_PROVIDER=azure`, missing `AZURE_LLM_KEY` or `AZURE_LLM_MODEL`: status disables Gemini and does not expose the primary Azure LLM dependency as misconfigured.
- `ENABLE_NEWS_MONITOR=true`, `MODEL_PROVIDER=openrouter`, missing `OPENROUTER_API_KEY`: status has the same false-ready gap for OpenRouter.

This can cause monitors or deploy checks to accept a configuration where `/api/news-monitor` will fail at runtime.

## Minimal fix proposal

- In `src/controllers/status.js`, add provider-specific readiness for the primary news-monitor LLM path.
- For `MODEL_PROVIDER=gemini`, require the same inputs used by `llmCallv2()` (`GEMINI_API_KEY` and `GEMINI_MODEL_NAME`), or change `llmCallv2()` to consistently use the configured fallback model and mirror that in status.
- For `MODEL_PROVIDER=azure`, report a primary Azure LLM dependency when `ENABLE_NEWS_MONITOR=true`; require `AZURE_LLM_KEY` and `AZURE_LLM_MODEL`, with the endpoint default allowed.
- For `MODEL_PROVIDER=openrouter`, report a primary OpenRouter dependency when `ENABLE_NEWS_MONITOR=true`; require `OPENROUTER_API_KEY` and model.
- Add integration coverage for the false-green cases above in `tests/integration/status-endpoint.test.js`.

Related PR: #57.

### PR DESCRIPTION
**Pull request by francovp**
### Motivation
- `/healthcheck` currently only reports uptime, which is not enough to tell whether gated features and integrations can actually process alerts.
- Provide a cheap, machine-readable JSON endpoint that surfaces non-sensitive metadata, feature-flag state, enabled delivery channels, and dependency readiness for operational tooling and monitors.

### Description
- Add `src/controllers/status.js` to compute a JSON status payload with `service`, `featureFlags`, `deliveryChannels`, and `dependencies` sections.
- Register `GET /api/status` and `GET /api/capabilities` in `src/routes/index.js`.
- Report TradingView MCP readiness correctly for the documented default URL path and when market-scanner depends on MCP even if enrichment is disabled.
- Report Firestore readiness correctly for supported managed ADC runtimes without treating bare Google project-id env vars as proof of credentials.
- Add integration coverage for alias behavior, API-key protection, payload shape, secret redaction, market-scanner MCP readiness, and Firestore ADC edge cases.

### Testing
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

### References
- Closes #52
- Fixes #76
- CB-10
- [Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a13e732ffec8324a33ddbece0c2075d)

### CODE REVIEW

File: src/controllers/status.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	});
+	const tradingViewMcp = dependencyStatus({
+		enabled: tradingViewMcpEnabled,
+		configured: hasValue(process.env.TRADINGVIEW_MCP_URL || DEFAULT_TRADINGVIEW_MCP_URL),
+	});
+	const firestore = dependencyStatus({
+		enabled: firestoreEnabled,
+		configured:
+			hasValue(process.env.FIREBASE_SERVICE_ACCOUNT_JSON)
+			|| hasValue(process.env.GOOGLE_APPLICATION_CREDENTIALS),
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Addressed in `e7efb94`.

`dependencies.firestore.configured` now treats Google-managed ADC runtimes as configured (Cloud Run/App Engine/Functions signals), while keeping explicit credential env vars supported (`FIREBASE_SERVICE_ACCOUNT_JSON`, `GOOGLE_APPLICATION_CREDENTIALS`).

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats Firestore ADC on Google-managed runtimes as configured`
- full status endpoint suite passes locally: `npm test -- tests/integration/status-endpoint.test.js`.

File: src/controllers/status.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+					&& hasValue(process.env.WHATSAPP_API_KEY)
+					&& hasValue(process.env.WHATSAPP_CHAT_ID),
+			},
+			gemini: {
+				enabled: geminiEnabled,
+				configured: hasValue(process.env.GEMINI_API_KEY),
+			},
+			tradingViewMcp: {
+				enabled: tradingViewMcpEnabled,
+				configured: hasValue(process.env.TRADINGVIEW_MCP_URL),
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Confirmed and already addressed on branch in `a111731`.

`dependencies.tradingViewMcp.configured` now honors the documented default MCP URL fallback, so `ENABLE_TRADINGVIEW_MCP_ENRICHMENT=true` with an unset `TRADINGVIEW_MCP_URL` is reported as configured/ready.

File: src/controllers/status.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+			&& hasValue(process.env.WHATSAPP_API_KEY)
+			&& hasValue(process.env.WHATSAPP_CHAT_ID),
+	});
+	const gemini = dependencyStatus({
+		enabled: geminiEnabled,
+		configured: hasValue(process.env.GEMINI_API_KEY),
+	});
+	const tradingViewMcp = dependencyStatus({
+		enabled: tradingViewMcpEnabled,
+		configured: hasValue(process.env.TRADINGVIEW_MCP_URL || DEFAULT_TRADINGVIEW_MCP_URL),
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js:210]: Fixed in `ed45fc2`.

`dependencies.tradingViewMcp.enabled` now turns on when either `ENABLE_TRADINGVIEW_MCP_ENRICHMENT` or `ENABLE_MARKET_SCANNER` is true, while `featureFlags.tradingViewMcpEnrichment` still reflects only the enrichment flag.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats TradingView MCP as enabled when market scanner depends on it`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+function isGoogleManagedRuntime() {
+	return (
+		hasValue(process.env.K_SERVICE)
+		|| hasValue(process.env.K_REVISION)
+		|| hasValue(process.env.FUNCTION_TARGET)
+		|| hasValue(process.env.FUNCTION_NAME)
+		|| hasValue(process.env.GOOGLE_CLOUD_PROJECT)
+		|| hasValue(process.env.GCP_PROJECT)
+		|| hasValue(process.env.GCLOUD_PROJECT)
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `5f839e1`.

`isGoogleManagedRuntime()` no longer treats `GOOGLE_CLOUD_PROJECT`, `GCP_PROJECT`, or `GCLOUD_PROJECT` as managed-runtime signals, so Firestore readiness only flips to ready for explicit credentials or runtime markers that actually imply ADC support.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `does not treat a bare Google project id as Firestore ADC readiness`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

**Comment by francovp** [comment]
Preview verification on `5f839e1` after the final status-endpoint fixes:

- GitHub Actions `build`: success
- Preview URL: `https://cabros-crypto-bot-telegram-pr-57.onrender.com`

Live probes against the deployed preview:
- `GET /healthcheck` -> `200`
- `GET /api/status` -> `200`
- `GET /api/capabilities` -> `200`
- Both status endpoints report `service.commit = 5f839e11ff0fec81e106735e54622b04efc4a71f`
- Live payload now shows `featureFlags.marketScanner = true` and `dependencies.tradingViewMcp.status = "ready"`
- Live payload also shows `dependencies.firestore.status = "misconfigured"`, which matches the follow-up fix that stops treating bare Google project-id env vars as ADC readiness

Remaining preview observation: `/api/status` and `/api/capabilities` are reachable without an API key here, so `WEBHOOK_API_KEY` still appears unset in this Render preview environment.

File: src/controllers/status.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `68f4f0c`.

`dependencies.gemini.enabled` now turns on when either `ENABLE_GEMINI_GROUNDING` or `ENABLE_NEWS_MONITOR` is true, while `featureFlags.geminiGrounding` still reflects only the alert-grounding flag.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats Gemini as enabled when news monitor depends on it`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

**Comment by francovp** [comment]
Final preview retest on `68f4f0c` after the Gemini/news-monitor readiness follow-up:

- GitHub Actions `build`: success
- Preview URL: `https://cabros-crypto-bot-telegram-pr-57.onrender.com`
- `GET /healthcheck` -> `200`
- `GET /api/status` -> `200`
- `GET /api/capabilities` -> `200`
- Both status endpoints now report `service.commit = 68f4f0cd964bfd83e37177048316e98795551e9b`
- Live payload still shows `dependencies.tradingViewMcp.status = "ready"`, `dependencies.gemini.status = "ready"`, and `dependencies.firestore.status = "misconfigured"` in this preview
- All PR review threads are resolved

Remaining preview observation: `WEBHOOK_API_KEY` still appears unset here because the status endpoints remain reachable without an API key in this Render preview environment.

File: src/controllers/status.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		},
+		dependencies: {
+			telegram,
+			whatsapp,
+			gemini,
+			tradingViewMcp,
+			firestore,
+			sentry,
+			langfuse,
+		},
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js:284]: Fixed in `641577c`.

`/api/status` now exposes `dependencies.llmAlertEnrichment`, and it only turns `enabled` on when the enrichment flag is set on an enabled news-monitor path. Configuration readiness follows the same Azure requirements enforced by `EnrichmentService.isEnabled()` (`AZURE_LLM_ENDPOINT`, `AZURE_LLM_KEY`, `AZURE_LLM_MODEL`).

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `reports Azure LLM enrichment readiness when the feature flag is enabled`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+		configured: hasValue(process.env.SENTRY_DSN),
+	});
+	const langfuse = dependencyStatus({
+		enabled: langfusePromptsEnabled,
+		configured: hasValue(process.env.LANGFUSE_PUBLIC_KEY) && hasValue(process.env.LANGFUSE_SECRET_KEY),
+	});
+	const llmAlertEnrichment = dependencyStatus({
+		enabled: llmAlertEnrichmentDependencyEnabled,
+		configured:
+			hasValue(process.env.AZURE_LLM_ENDPOINT)
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `a2f4d29`.

`dependencies.llmAlertEnrichment.configured` now honors the same default endpoint path as the shared Azure client config by treating the GitHub Models inference fallback (`https://models.github.ai/inference`) as configured when `AZURE_LLM_ENDPOINT` is unset.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats the default Azure LLM endpoint as configured for enrichment readiness`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+	const whatsapp = dependencyStatus({
+		enabled: whatsappEnabled,
+		configured:
+			hasValue(process.env.WHATSAPP_API_URL)
+			&& hasValue(process.env.WHATSAPP_API_KEY)
+			&& hasValue(process.env.WHATSAPP_CHAT_ID),
+	});
+	const gemini = dependencyStatus({
+		enabled: geminiEnabled,
+		configured: hasValue(process.env.GEMINI_API_KEY),
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `5fd6317`.

`dependencies.gemini` now only turns on for `news-monitor` when the active provider path actually requires Gemini. That keeps the default Gemini path covered, but stops flagging supported Brave-search plus Azure/OpenRouter deployments as misconfigured when `GEMINI_API_KEY` is intentionally absent.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `does not require Gemini when news monitor uses Brave search and Azure`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

**Comment by francovp** [comment]
Daily bug scan found one concrete readiness gap in this PR and opened #76.

Short version: the new status payload tracks Gemini readiness for `ENABLE_NEWS_MONITOR`, but it does not fully mirror the provider dispatch in `genaiClient.llmCallv2()`:

- `MODEL_PROVIDER=gemini` needs `GEMINI_MODEL_NAME` as well as `GEMINI_API_KEY`, otherwise `llmCallv2()` skips Gemini.
- `MODEL_PROVIDER=azure` / `openrouter` need their own primary news-monitor readiness, not only the optional `llmAlertEnrichment` dependency.

Minimal PR-local fix should be in `src/controllers/status.js` plus `tests/integration/status-endpoint.test.js`: add provider-specific readiness for the primary news-monitor LLM path and cover the false-green cases from #76.

**Comment by francovp** [comment]
Follow-up for #76 is now on PR head `ddc18d0`.

What changed:
- added `dependencies.newsMonitorLlm` to `/api/status` and `/api/capabilities` so the primary news-monitor LLM provider is reported explicitly (`gemini`, `azure`, or `openrouter`)
- kept Gemini/search readiness separate from primary provider readiness
- covered the false-green cases from #76 in `tests/integration/status-endpoint.test.js`
- updated the README status payload example

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`
- GitHub Actions `Node.js CI` run `#158` passed for `ddc18d0`
- Preview `https://cabros-crypto-bot-telegram-pr-57.onrender.com` is serving commit `ddc18d02075a65b1f734e53ac17d2bab80497b55`
- Live preview checks:
  - `GET /healthcheck` -> `200`
  - `GET /api/status` -> `200`
  - `GET /api/capabilities` remains aliased to the same payload shape

No unresolved review threads are open right now.

File: src/controllers/status.js [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+function getStatus() {
+	const previewEnvironment = isRenderPreview();
+	const modelProvider = getModelProvider();
+	const telegramFlagEnabled = isEnabled(process.env.ENABLE_TELEGRAM_BOT);
+	const telegramEnabled = telegramFlagEnabled && !previewEnvironment;
+	const whatsappEnabled = isEnabled(process.env.ENABLE_WHATSAPP_ALERTS);
+	const geminiGroundingEnabled = isEnabled(process.env.ENABLE_GEMINI_GROUNDING);
+	const newsMonitorEnabled = isEnabled(process.env.ENABLE_NEWS_MONITOR);
+	const forceBraveSearch = isEnabled(process.env.FORCE_BRAVE_SEARCH);
+	const newsMonitorUsesGeminiSearch = newsMonitorEnabled && !forceBraveSearch;
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js:179]: Fixed in `d68df84`.

`/api/status` now exposes `dependencies.braveSearch` for the forced Brave Search path, so `ENABLE_NEWS_MONITOR=true` plus `FORCE_BRAVE_SEARCH=true` reports `misconfigured` unless `BRAVE_SEARCH_API_KEY` is present. That keeps the provider-specific `newsMonitorLlm` readiness intact while covering the search leg that `genaiClient.search()` routes through in this mode.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `does not require Gemini when news monitor uses Brave search and Azure`
- `reports forced Brave search as misconfigured when its API key is missing`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+		|| hasValue(process.env.FUNCTION_TARGET)
+		|| hasValue(process.env.FUNCTION_NAME)
+		|| hasValue(process.env.GAE_SERVICE)
+	);
+}
+
+function getModelProvider() {
+	return typeof process.env.MODEL_PROVIDER === 'string' && process.env.MODEL_PROVIDER.trim().length > 0
+		? process.env.MODEL_PROVIDER.trim().toLowerCase()
+		: 'gemini';
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js:65]: Fixed in `2679b81`.

`MODEL_PROVIDER` is now normalized in the shared grounding config before `genaiClient.llmCallv2()` dispatches, so mixed-case values like `Azure` resolve to the same lowercase provider that `/api/status` reports.

Added coverage for both layers:
- `tests/unit/genai-client-provider-normalization.test.js`: `treats mixed-case MODEL_PROVIDER values as Azure at runtime`
- `tests/integration/status-endpoint.test.js`: `normalizes mixed-case primary news monitor provider names`

Verification:
- `pnpm test -- tests/unit/genai-client-provider-normalization.test.js --runInBand`
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+	});
+	const tradingViewMcp = dependencyStatus({
+		enabled: tradingViewMcpEnabled,
+		configured: hasValue(process.env.TRADINGVIEW_MCP_URL || DEFAULT_TRADINGVIEW_MCP_URL),
+	});
+	const firestore = dependencyStatus({
+		enabled: firestoreEnabled,
+		configured:
+			hasValue(process.env.FIREBASE_SERVICE_ACCOUNT_JSON)
+			|| hasValue(process.env.GOOGLE_APPLICATION_CREDENTIALS)
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `2679b81`.

`dependencies.firestore` no longer treats `FIREBASE_SERVICE_ACCOUNT_JSON` as configured just because the env var is non-empty. The status check now requires that inline credentials parse as a JSON object before marking the Firestore path ready, which matches `AlertStorageService.getFirestore()` more closely.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats malformed inline Firestore credentials as misconfigured`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+		configured: hasValue(process.env.GEMINI_API_KEY),
+	});
+	const tradingViewMcp = dependencyStatus({
+		enabled: tradingViewMcpEnabled,
+		configured: hasValue(process.env.TRADINGVIEW_MCP_URL || DEFAULT_TRADINGVIEW_MCP_URL),
+	});
+	const firestore = dependencyStatus({
+		enabled: firestoreEnabled,
+		configured:
+			hasValidJsonObject(process.env.FIREBASE_SERVICE_ACCOUNT_JSON)
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in `42d8d67`.

`dependencies.firestore` now requires the inline credential payload to expose the same core service-account fields Firebase Admin consumes (`project_id`, `client_email`, `private_key`) and for the private key to parse as PEM before `/api/status` reports the Firestore path as ready.

Added integration coverage in `tests/integration/status-endpoint.test.js`:
- `treats malformed inline Firestore credentials as misconfigured`
- `treats incomplete inline Firestore credentials as misconfigured`

Verification:
- `pnpm test -- tests/integration/status-endpoint.test.js --runInBand`
- `pnpm test --runInBand`

File: src/controllers/status.js [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+	const whatsapp = dependencyStatus({
+		enabled: whatsappEnabled,
+		configured:
+			hasValue(process.env.WHATSAPP_API_URL)
+			&& hasValue(process.env.WHATSAPP_API_KEY)
+			&& hasValue(process.env.WHATSAPP_CHAT_ID),
+	});
+	const gemini = dependencyStatus({
+		enabled: geminiEnabled,
+		configured: hasValue(process.env.GEMINI_API_KEY),
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in fa122f5.

When ENABLE_GEMINI_GROUNDING=true and the active provider path is Gemini, dependencies.gemini now requires both GEMINI_API_KEY and GEMINI_MODEL_NAME, matching the genaiClient.llmCallv2() gate used during grounded alert generation. The earlier split for newsMonitorLlm stays intact, so search-only Gemini paths still report independently from the provider-specific LLM dependency.

Added integration coverage in tests/integration/status-endpoint.test.js:
- treats Gemini grounding as misconfigured without a Gemini model on the Gemini provider path

Verification:
- pnpm test -- tests/integration/status-endpoint.test.js --runInBand
- pnpm test --runInBand

File: src/controllers/status.js [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+	});
+	const tradingViewMcp = dependencyStatus({
+		enabled: tradingViewMcpEnabled,
+		configured: hasValue(process.env.TRADINGVIEW_MCP_URL || DEFAULT_TRADINGVIEW_MCP_URL),
+	});
+	const firestore = dependencyStatus({
+		enabled: firestoreEnabled,
+		configured:
+			hasValidInlineFirestoreCredentials(process.env.FIREBASE_SERVICE_ACCOUNT_JSON)
+			|| hasValue(process.env.GOOGLE_APPLICATION_CREDENTIALS)
```
[the remark this answers is not in the document]
Comment (francovp) [reply src/controllers/status.js]: Fixed in c26ca4e.

Firestore readiness no longer treats GOOGLE_APPLICATION_CREDENTIALS as configured just because the env var is non-empty. /api/status now requires the referenced file to exist and be readable, which matches the fail-open initialization path in AlertStorageService much more closely for dead credential-file paths.

Added integration coverage in tests/integration/status-endpoint.test.js:
- treats an unreadable Firestore credential file path as misconfigured
- treats a readable Firestore credential file path as configured

Verification:
- pnpm test -- tests/integration/status-endpoint.test.js --runInBand
- pnpm test --runInBand
